### PR TITLE
Fixing APIMANAGER-3948 - Moving synapse msgContext api properties setting code into auth handler

### PR DESCRIPTION
**Pull request by malinthaprasan**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java [hunk 1]
```diff
@@ -262,4 +267,91 @@ private String logMessageDetails(MessageContext messageContext, long elapsedTime
...
+        String applicationId = "";
+        if (authContext != null) {
+            consumerKey = authContext.getConsumerKey();
+            username = authContext.getUsername();
+            applicationName = authContext.getApplicationName();
+            applicationId = authContext.getApplicationId();
+        }
+
+        String context = (String) messageContext.getProperty(RESTConstants.REST_API_CONTEXT);
+        String api_version = (String) messageContext.getProperty(RESTConstants.SYNAPSE_REST_API);
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java]: use camel case for variable as for maintain the consistancy

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java [hunk 2]
```diff
@@ -262,4 +267,91 @@ private String logMessageDetails(MessageContext messageContext, long elapsedTime
...
+        String resource = "/";
+        Pattern pattern = Pattern.compile("^/.+?/.+?([/?].+)$");
+        Matcher matcher = pattern.matcher((String) mc.getProperty(RESTConstants.REST_FULL_REQUEST_PATH));
+        if (matcher.find()) {
+            resource = matcher.group(1);
+        }
+        return resource;
+    }
+
+    private String getAPIProviderFromRESTAPI(String api_version) {
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java]: use camel case for variable as for maintain the consistancy

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java [hunk 3]
```diff
@@ -262,4 +267,91 @@ private String logMessageDetails(MessageContext messageContext, long elapsedTime
...
+        messageContext.setProperty(APIMgtGatewayConstants.HTTP_METHOD, method);
+        messageContext.setProperty(APIMgtGatewayConstants.HOST_NAME, hostName);
+        messageContext.setProperty(APIMgtGatewayConstants.API_PUBLISHER, apiPublisher);
+        messageContext.setProperty(APIMgtGatewayConstants.APPLICATION_NAME, applicationName);
+        messageContext.setProperty(APIMgtGatewayConstants.APPLICATION_ID, applicationId);
+    }
+
+    private String extractResource(MessageContext mc) {
+        String resource = "/";
+        Pattern pattern = Pattern.compile("^/.+?/.+?([/?].+)$");
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java]: Better to use constants

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java (hunk 1 above)
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java]: Fixed in #1494 (https://github.com/wso2/carbon-apimgt/commit/68b378607966da79d994692e05c46be213f2ecc0)

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java (hunk 2 above)
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java]: Fixed in #1494 (https://github.com/wso2/carbon-apimgt/commit/68b378607966da79d994692e05c46be213f2ecc0)

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java (hunk 3 above)
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java]: Fixed in #1494 (https://github.com/wso2/carbon-apimgt/commit/68b378607966da79d994692e05c46be213f2ecc0)

**Pull request merged**